### PR TITLE
feat(youtube): `copy-video-url` patch

### PR DESCRIPTION
**Pull request by aliernfrog**
Depends on https://github.com/revanced/revanced-patches/pull/1401

Should be merged with https://github.com/revanced/revanced-integrations/pull/263

Related issue: https://github.com/revanced/revanced-patches/issues/531

Adds copy video & copy timestamp URL buttons in video player.
IMG_20221230_234907

### CODE REVIEW

File: src/main/kotlin/app/revanced/patches/youtube/interaction/copyvideourl/bytecode/patch/CopyVideoUrlBytecodePatch.kt [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    }
+
+    override fun execute(context: BytecodeContext): PatchResult {
+        // Hook video time
+        with(VideoInformationPatch) {
+            videoTimeHook(
+                INTEGRATIONS_PATCH_UTILS_DESCRIPTOR,
+                "setVideoTime"
+            )
+        }
```
Comment (oSumAtrIX) [inline src/main/kotlin/app/revanced/patches/youtube/interaction/copyvideourl/bytecode/patch/CopyVideoUrlBytecodePatch.kt:47]: The function `with` is not necessary here as the new scope is only used for a single function.

File: src/main/kotlin/app/revanced/patches/youtube/interaction/copyvideourl/bytecode/patch/CopyVideoUrlBytecodePatch.kt [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+        // Inject call for video id
+        VideoIdPatch.injectCall("$INTEGRATIONS_PATCH_UTILS_DESCRIPTOR->setVideoId(Ljava/lang/String;)V")
```
Comment (oSumAtrIX) [inline src/main/kotlin/app/revanced/patches/youtube/interaction/copyvideourl/bytecode/patch/CopyVideoUrlBytecodePatch.kt:50]: This is not necessary. There is another patch that can be used as a dependency for this one. The integrations of that patch contain a getter for the current video id. The only time `VideoIdPatch` is required is if you need to update something based on a new video or its id. For example, the `return-youtube-dislike` requires it, in order to update the dislike count. This PR adds buttons that only require to use of the mentioned getter for the video id.

**Review by oSumAtrIX — CHANGES_REQUESTED** [review]
Looks good overall! I have addressed some small changes as well as in the depending integrations PR.

**Comment by Domiiniik** [comment]
Can showing/hiding both buttons can be configured so I can for instance hide copy video button but leave copy timestamp button? Vanced did allowed this.

**Comment by oSumAtrIX** [comment]
That is controlled by the settings patch.

File: src/main/kotlin/app/revanced/patches/youtube/interaction/copyvideourl/bytecode/patch/CopyVideoUrlBytecodePatch.kt (hunk 2 above)
Comment (aliernfrog) [reply src/main/kotlin/app/revanced/patches/youtube/interaction/copyvideourl/bytecode/patch/CopyVideoUrlBytecodePatch.kt:48]: Integrations of some patches (RYD, remember video quality and downloads) has getter method for `videoId`, but I couldn't find any other patch this patch can depend on.

Comment (oSumAtrIX) [reply src/main/kotlin/app/revanced/patches/youtube/interaction/copyvideourl/bytecode/patch/CopyVideoUrlBytecodePatch.kt:48]: What I meant is that the `VideoInformation` patch should include a getter for the videoId. The `downloads` and the patch from this PR would then make use of it. For that the field `videoId` from the `downloads` patch should be moved to
 `app.revanced.integrations.patches.VideoInformation`. Make sure the respective changes are also reflected in the `VideoInformation` patch (such as adding `VideoIdPatch.injectCall` to it and making the `downloads` and your patch dependant on the `VideoInformation` patch).

**Comment by aliernfrog** [comment]
Made following changes:
- Removed unneeded `with` function
- Added video id to VideoInformation patch and used that in copy video url and downloads patch
- Some changes from integrations PR
- Updated icons
IMG_20221231_163653

**Comment by Indranil012** [comment]
>       IMG_20221231_163653

Icons are looking bigger and bolder than download and full screen icon

File: src/main/kotlin/app/revanced/patches/youtube/interaction/copyvideourl/bytecode/patch/CopyVideoUrlBytecodePatch.kt [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+            "$INTEGRATIONS_PLAYER_PACKAGE/CopyVideoUrlTimestampButton;"
+        )
+    }
+
+    override fun execute(context: BytecodeContext): PatchResult {
+        // Hook video time
+        VideoInformationPatch.videoTimeHook(
+            INTEGRATIONS_PATCH_UTILS_DESCRIPTOR,
+            "setVideoTime"
+        )
```
Comment (oSumAtrIX) [inline src/main/kotlin/app/revanced/patches/youtube/interaction/copyvideourl/bytecode/patch/CopyVideoUrlBytecodePatch.kt:43]: Is there a specific reason to do that? Can the integrations not call `VideoInformationPatch.getVideoTime` in the integrations?

**Review by oSumAtrIX — CHANGES_REQUESTED** [review]

**Comment by oSumAtrIX** [comment]
Thanks!

**Pull request merged**